### PR TITLE
Fixed 4 broken links

### PR DESCRIPTION
**Pull request by thinksanky**
## Description ##
* Fixed the remaining broken links reported on Feb 4th 2018

### CODE REVIEW

**Comment by thinksanky** [comment]
@eric-haibin-lin - please review and merge

File: python/mxnet/gluon/trainer.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ class Trainer(object):
         The set of parameters to optimize.
     optimizer : str or Optimizer
         The optimizer to use. See
-        `help <http://mxnet.io/api/python/optimization.html#the-mxnet-optimizer-package>`_
+        `help <http://mxnet.io/api/python/optimization/optimization.html#the-mxnet-optimizer-package>`_
         on Optimizer for a list of available optimizers.
```
Comment (szha) [inline python/mxnet/gluon/trainer.py]: Fix lint. Line is too long

File: docs/tutorials/index.md [hunk 2]
```diff
@@ -174,7 +174,7 @@ The Gluon and Module tutorials are in Python, but you can also find a variety of
 
 <div class="applications">
 
-- [Connectionist Temporal Classification](http://mxnet.incubator.apache.org/tutorials/speech_recognition/ctc.html)
+- [Connectionist Temporal Classification](../tutorials/speech_recognition/ctc.html)
```
Comment (eric-haibin-lin) [inline docs/tutorials/index.md]: Maybe we should use relative links to avoid version issues and keep it as a general guideline for all other links?

**Comment by eric-haibin-lin** [comment]
Please fix lint

File: docs/tutorials/index.md (hunk 2 above)
Comment (thinksanky) [reply docs/tutorials/index.md]: yes - this is a good practice. I will scan others and carefully do this in a different PR.

**Comment by eric-haibin-lin** [comment]
Can you also update the broken link in `docs/faq/multi_devices.md` for ps-lite to be http://ps-lite.readthedocs.io/en/latest/

**Pull request merged**